### PR TITLE
Sui system state support one push upgrade

### PR DESCRIPTION
**Pull request by lxfind**
This PR adds support for upgrading sui system state using only one framework upgrade.
It removes next_system_state_version from the parameters to advance_epoch, since we no longer use it to trigger upgrade.
Instead, inside the inner system state load function, we could upgrade it when we need to. Examples will come in latter in tests.
This PR also removes the default impl for SuiSystemState, since it's not needed.

### CODE REVIEW

File: crates/sui-framework/packages/sui-system/sources/sui_system.move [hunk 1]
```diff
@@ -534,79 +523,82 @@ module sui_system::sui_system {
 
     /// Return the current epoch number. Useful for applications that need a coarse-grained concept of time,
     /// since epochs are ever-increasing and epoch changes are intended to happen every 24 hours.
-    public fun epoch(wrapper: &SuiSystemState): u64 {
+    public fun epoch(wrapper: &mut SuiSystemState): u64 {
```
Comment (emmazzz) [inline crates/sui-framework/packages/sui-system/sources/sui_system.move:526]: It feels off to have a getter take in a mutable reference but I guess with this approach we don't have a way around it. Also this is a public fun so we won't be able to change its signature in any framework upgrade later. Let's make sure we want to do this.

Comment (emmazzz) [reply crates/sui-framework/packages/sui-system/sources/sui_system.move:526]: Actually I don't think these getters are used outside of tests right now. Do we want to make them test only and only upgrade to non test only if we need to?

Comment (lxfind) [reply crates/sui-framework/packages/sui-system/sources/sui_system.move:526]: My general question/thought is how much interface we want to provide for dapp developers who wants to interact with system state in their contracts

**Review by amnn — APPROVED** [review]
LGTM!  I do wonder if it's going to get strange to see all the APIs for system state accept it by mutable reference -- certainly it'll be harder to distinguish reads from writes.

**Pull request merged**